### PR TITLE
Add PM support for EFR32BG SoCs

### PR DESCRIPTION
**Pull request by fkokosinski**
This PR adds PM support for EFR32BG SoCs. Changes introduced in this PR were tested on the in-tree `efr32bg_sltb010a` board.

### CODE REVIEW

File: drivers/serial/uart_gecko.c [hunk 1]
```diff
@@ -252,7 +259,11 @@ static int uart_gecko_irq_tx_complete(const struct device *dev)
 
 	USART_IntClear(config->base, USART_IF_TXC);
 
-	return (flags & USART_IF_TXC) != 0U;
+	if ((flags & USART_IF_TXC) != 0U) {
+		pm_policy_state_lock_put(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
+		return 1;
```
Comment (tgorochowik) [inline drivers/serial/uart_gecko.c:264]: `-EINVAL`? Or other errno if something fits better

File: soc/arm/silabs_exx32/common/soc_power.c [hunk 2]
```diff
@@ -63,3 +151,69 @@ __weak void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 	ARG_UNUSED(state);
 	ARG_UNUSED(substate_id);
 }
+
+#ifdef CONFIG_SOC_GECKO_DEV_INIT
+
+/* Function to check any ready threads are scheduled */
+bool ConfirmSleepModeStatus(void)
```
Comment (tgorochowik) [inline soc/arm/silabs_exx32/common/soc_power.c:158]: Why the CamelCase?

**Review by tgorochowik — CHANGES_REQUESTED** [review]

File: soc/arm/silabs_exx32/common/soc.c [hunk 3]
```diff
@@ -230,4 +250,13 @@ static int silabs_exx32_init(const struct device *arg)
 	return 0;
 }
 
+#if defined(CONFIG_SOC_GECKO_DEV_INIT) && defined(CONFIG_PM)
+void init_hfxo_irq(void)
+{
+	IRQ_DIRECT_CONNECT(HFXO0_IRQn, 2, sl_hfxo_manager_irq_handler, 0);
```
Comment (tgorochowik) [inline soc/arm/silabs_exx32/common/soc.c:256]: Can we add the interrupt info to the devicetree here and then use the `DT_INST_IRQN` and `DT_INST_IRQN` macros?

File: soc/arm/silabs_exx32/common/soc_power.c (hunk 2 above)
Comment (fkokosinski) [reply soc/arm/silabs_exx32/common/soc_power.c]: I've refactored the code and this piece of code is no longer needed.

File: soc/arm/silabs_exx32/common/soc.c (hunk 3 above)
Comment (fkokosinski) [reply soc/arm/silabs_exx32/common/soc.c]: I've refactored the code and this piece of code is no longer needed.

**Review by tgorochowik — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**